src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts

```diff
@@ -107,6 +117,56 @@ export class NotificationServiceModalComponent implements OnChanges {
         });
   }
 
+  monitorNotificationServiceInstallationStatus(data: any, pluginName: string) {
+    this.service.monitorPluginInstallationStatus(data.statusLink)
+      .pipe(
+        take(1),
+        // checking the response object for plugin.
+        // if pacakge.status === 'in-progress' then
+        // throw an error to re-fetch:
+        tap((response: any) => {
+          if (response.packageStatus[0].status === 'in-progress') {
+            this.increment++;
+            throw response;
+          }
+        }),
+        retryWhen(result =>
+          result.pipe(
+            // only if a server returned an error, stop trying and pass the error down
+            tap(installStatus => {
+              if (installStatus.error) {
+                this.ngProgress.done();
+                this.alertService.closeMessage();
+                throw installStatus.error;
+              }
+            }),
+            delayWhen(() => {
+              const delay = this.increment * this.initialDelay;     // incremental
+              console.log(new Date().toLocaleString(), `retrying after ${delay} msec...`);
+              return timer(delay);
+            }), // delay between api calls
+            // Set the number of attempts.
+            take(this.maxRetry),
+            // Throw error after exceed number of attempts
+            concatMap(o => {
+              if (this.increment > this.maxRetry) {
+                this.ngProgress.done();
+                this.alertService.closeMessage();
+                // tslint:disable-next-line: max-line-length
+                return throwError(`Failed to get expected results in ${this.maxRetry} attempts, tried with incremental time delay starting with 2s, for installing plugin ${pluginName}`);
+              }
+              return of(o);
+            }),
+          ))
+      ).subscribe(() => {
+        this.ngProgress.done();
+        this.alertService.closeMessage();
+        this.alertService.success('Notification service added successfully.', true);
```

Comment:
plugin vs service, We should show message as per type. 
Also looks like the last delay/retry interval is being added before get installed call after status success / error of status.